File: src/main/java/com/tiobe/plugins/intellij/actions/AnalyzeFile.java
package com.tiobe.plugins.intellij.actions;

import com.intellij.execution.configurations.GeneralCommandLine;
import com.intellij.openapi.project.Project;
import com.intellij.openapi.vfs.VirtualFile;

public class AnalyzeFile extends AbstractAnalyzeAction {
    public GeneralCommandLine getTICSCommand(VirtualFile file, Project project) {
        GeneralCommandLine cmd = new GeneralCommandLine(TICS_COMMAND);
        cmd.addParameters(IDE_PARAMETER, IDE_NAME);
        final VirtualFile projectFile = project == null ? null : project.getProjectFile();
        if (projectFile != null) {
            cmd.addParameters(PROJFILE_PARAMETER, projectFile.getCanonicalPath());
        }
        if (file != null && file.getCanonicalPath() != null) {
            cmd.addParameter(file.getCanonicalPath());
        }
        return cmd;
    }

    @Override
    public boolean isFileRequired() {
        return true;
    }

    @Override
    public boolean isProjectRequired() {
        return false;
    }
}

File: src/main/kotlin/com/tiobe/plugins/intellij/actions/AnalyzeFile.kt
package com.tiobe.plugins.intellij.actions

import com.intellij.execution.configurations.GeneralCommandLine
import com.intellij.openapi.project.Project
import com.intellij.openapi.vfs.VirtualFile

open class AnalyzeFile : AnalyzeAction() {
    override fun getTicsCommand(file: VirtualFile?, project: Project?): GeneralCommandLine {
        val command = GeneralCommandLine(TICS_COMMAND)
        command.addParameters(IDE_PARAMETER, IDE_NAME)

        project?.projectFile?.canonicalPath?.let {
            command.addParameters(PROJECT_FILE_PARAMETER, it)
        }

        file?.canonicalPath?.let {
            command.addParameter(it)
        }

        return command
    }

    override fun isFileRequired(): Boolean {
        return true
    }

    override fun isProjectRequired(): Boolean {
        return false
    }
}
